Lib/inspect.py

```diff
@@ -313,7 +314,9 @@ def ismethoddescriptor(object):
         # mutual exclusion
         return False
     tp = type(object)
-    return hasattr(tp, "__get__") and not hasattr(tp, "__set__")
+    if hasattr(tp, "__set__") or hasattr(tp, "__delete__"):
+        return False
+    return hasattr(tp, "__get__")
```

Comment:
I would check `__get__` first. It looks more natural.
```suggestion
    return hasattr(tp, "__get__") and not (hasattr(tp, "__set__") or hasattr(tp, "__delete__"))
```

Reply:
OK, changed. :)